expat/configure.ac

```diff
@@ -94,6 +94,9 @@ AC_LANG_POP([C++])
 AS_IF([test "$GCC" = yes],
   [AX_APPEND_LINK_FLAGS([-fno-strict-aliasing],[LDFLAGS])])
 
+AS_IF([test "$GCC" = yes],
+      archive_cmds='$CC -shared $pic_flag -o $lib $libobjs $deplibs $compiler_flags')
```

Comment:
That's an interesting idea!

I have compared this now to what `configure` puts into `libtool` for `LD=ld`:

```
# LD=ld
\$CC -shared \$pic_flag          \$libobjs \$deplibs \$compiler_flags \$wl-soname \$wl\$soname -o \$lib

# Proposed in this pull request
\$CC -shared \$pic_flag -o \$lib \$libobjs \$deplibs \$compiler_flags
```
I noticed the absense of `\$wl-soname \$wl\$soname` as well as a different location of `-o \$lib`. Is that intended?

I tried former command with `CC=clang LD=clang++` and it worked. My libtool is at 2.4.6 over here if it matters.